### PR TITLE
Bugfix/consensus/duplicate messages

### PR DESCRIPTION
**Pull request by liayoo**
- Changed to prevent processing same consensus messages more than once
- Adjust and sync epoch every 100 epoch

### CODE REVIEW

File: consensus/index.js [hunk 1]
```diff
@@ -92,9 +94,18 @@ class Consensus {
     if (this.epochInterval) {
       clearInterval(this.epochInterval);
     }
-    this.epochInterval = setInterval(() => {
+    this.epochInterval = setInterval(async () => {
       this.tryFinalize();
       let currentTime = Date.now();
+      if (this.state.epoch % 100 === 0) {
+        // adjust time
+        const iNTPData = await ntpsync.ntpLocalClockDeltaPromise();
```
Comment (minsulee2) [inline consensus/index.js:102]: What about try {} catch { return 0 } block for this?
If ntp fails, it may cause epoch corruption.

Comment (liayoo) [reply consensus/index.js]: Thanks, added a try-catch block.

**Comment by liayoo** [comment]
Addressed all comments, PTAL @minsulee2

File: consensus/index.js (hunk 1 above)
Comment (minsulee2) [reply consensus/index.js]: Coooooool. Thanks for the update!

**Pull request merged**